lib/rust/web/src/lib.rs

```diff
@@ -947,3 +948,52 @@ impl TimeProvider for Performance {
         self.now()
     }
 }
+
+
+// ====================
+// === FrameCounter ===
+// ====================
+
+type Counter = Rc<std::cell::Cell<i32>>;
+
+#[derive(Debug)]
+/// A counter that counts frames thr frames that have passed since its initialization.
+///
+/// Uses `request_animation_frame` under the hood to count frames.
+pub struct FrameCounter {
+    frames:                Counter,
+    js_on_frame_handle_id: i32,
+    _closure_handle:       Rc<RefCell<Option<Closure<(dyn FnMut(f64))>>>>,
+}
+
+impl FrameCounter {
+    /// Creates a new frame counter.
+    pub fn start_counting() -> Self {
+        let frames: Counter = default();
+        let frames_handle = Rc::clone(&frames);
+        let closure_handle = Rc::new(RefCell::new(None));
+        let closure_handle_internal = Rc::clone(&closure_handle);
```

Comment:
Is there a memory leak here? The closure strongly-references the closure handle, which owns the closure.